model/DataObject.php

```diff
@@ -2677,8 +2677,7 @@ public function setField($fieldName, $val) {
 				user_error('DataObject::setField: passed an object that is not a DBField', E_USER_WARNING);
 			}
 
-			$dbField = $this->dbObject($fieldName);
-			if ($dbField && $dbField->scalarValueOnly() && !empty($val) && !is_scalar($val)){
+			if ($this->db($fieldName) && ($dbField = $this->dbObject($fieldName))  && $dbField->scalarValueOnly() && !empty($val) && !is_scalar($val)){
```

Comment:
Can we reduce the line length here? I believe 120 is our hard-limit
```suggestion
			$dbField = ($this->db($fieldName)) ? $this->dbObject($fieldName) : null;
			if ($dbField && $dbField->scalarValueOnly() && !empty($val) && !is_scalar($val)){
```